tempodb/encoding/vparquet/schema_test.go

```diff
@@ -22,8 +24,90 @@ func TestProtoParquetRoundTrip(t *testing.T) {
 	// Proto -> Parquet -> Proto
 
 	traceIDA := []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F}
+	expectedTrace := allFieldsTrace(traceIDA)
 
-	expectedTrace := &tempopb.Trace{
+	parquetTrace := traceToParquet(traceIDA, expectedTrace, nil)
+	actualTrace := parquetTraceToTempopbTrace(parquetTrace)
+	assert.Equal(t, expectedTrace, actualTrace)
+}
+
+func TestProtoToParquetEmptyTrace(t *testing.T) {
+	want := &Trace{
+		TraceID:       make([]byte, 16),
+		ResourceSpans: nil,
+	}
+
+	got := traceToParquet(nil, &tempopb.Trace{}, nil)
+	require.Equal(t, want, got)
+}
+
+func TestProtoParquetRando(t *testing.T) {
+	trp := &Trace{}
+	for i := 0; i < 100; i++ {
+		batches := rand.Intn(15)
+		id := test.ValidTraceID(nil)
+		expectedTrace := test.MakeTrace(batches, id)
+
+		parqTr := traceToParquet(id, expectedTrace, trp)
+		actualTrace := parquetTraceToTempopbTrace(parqTr)
+		require.Equal(t, expectedTrace, actualTrace)
+	}
+}
+
+func TestFieldsAreCleared(t *testing.T) {
+	traceID := []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F}
```

Comment:
Thinking about ways to ensure this test stays updated as we add fields to the `vparquet.Trace` struct.  There are similar fully-populated traces [here](https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet/block_traceql_test.go#L395), and [here](https://github.com/grafana/tempo/blob/main/tempodb/encoding/vparquet/block_search_test.go#L31) wonder if we can reuse by converting to tempopb.Trace.  That fully-populated traces should be kept up to date with new columns naturally by the search tests.

Reply:
- swapped the schema_tests to use `fullyPopulatedTestTrace()`
- removed allFieldsTrace
- extended the trace in `fullyPopulatedTestTrace()` to include more fields
- fixed a few latent bugs in `traceToParquet`